app/src/main/java/com/smlnskgmail/jaman/hashcheckerlite/MainActivity.java

```diff
@@ -210,13 +214,35 @@ public void onBackPressed() {
                 BaseFragment.CURRENT_FRAGMENT_TAG
         );
         if (fragment instanceof AppBackClickTarget) {
-            ((AppBackClickTarget) fragment).appBackClick();
+            waitForDoubleBackPressed();
         }
         for (Fragment fragmentInApp : getSupportFragmentManager().getFragments()) {
             if (fragmentInApp instanceof AppResumeTarget) {
                 ((AppResumeTarget) fragmentInApp).appResume();
             }
         }
+
+    }
+
+    private void waitForDoubleBackPressed() {
+        if (doubleBackToExitPressedOnce) {
+            Intent a = new Intent(Intent.ACTION_MAIN);
+            a.addCategory(Intent.CATEGORY_HOME);
+            a.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+            startActivity(a);
+            return;
+        }
+
+        this.doubleBackToExitPressedOnce = true;
+        Toast.makeText(this, "Please click BACK again to exit", Toast.LENGTH_SHORT).show();
```

Comment:
Extract text in toast in string resources. I'm translate this sentence to other languages.